src/CSESoftware.Repository/IRepository.cs

```diff
@@ -1,22 +1,37 @@
-﻿using System.Threading.Tasks;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
 using CSESoftware.Core.Entity;
 
 namespace CSESoftware.Repository
 {
     public interface IRepository : IReadOnlyRepository
     {
         void Create<TEntity>(TEntity entity)
-            where TEntity : class, IBaseEntity;
+            where TEntity : class, IEntity;
+
+        void Create<TEntity>(List<TEntity> entities)
+            where TEntity : class, IEntity;
 
         void Update<TEntity>(TEntity entity)
-            where TEntity : class, IBaseEntity;
+            where TEntity : class, IEntity;
+
+        void Update<TEntity>(List<TEntity> entities)
+            where TEntity : class, IEntity;
 
-        void Delete<TEntity>(object id)
-            where TEntity : class, IBaseEntity;
+        void Delete<TEntity, T>(T id)
```

Comment:
What does the implementation/usage for this look like?

Reply:
_repository.Delete<Pizza, int>(1);

Reply:
Is that better than _repository.Delete<Pizza>(1)? What value do we get from specifying the type? I have not looked at implementation yet so maybe it's used there.

Reply:
This is do to Core now taking the type of an ID rather than just making it object